coderd/workspacebuilds.go

```diff
@@ -288,28 +288,10 @@ func (api *API) workspaceBuildByBuildNumber(rw http.ResponseWriter, r *http.Requ
 	httpapi.Write(ctx, rw, http.StatusOK, apiBuild)
 }
 
-// Azure supports instance identity verification:
-// https://docs.microsoft.com/en-us/azure/virtual-machines/windows/instance-metadata-service?tabs=linux#tabgroup_14
-//
-// @Summary Create workspace build
-// @ID create-workspace-build
-// @Security CoderSessionToken
-// @Accept json
-// @Produce json
-// @Tags Builds
-// @Param workspace path string true "Workspace ID" format(uuid)
-// @Param request body codersdk.CreateWorkspaceBuildRequest true "Create workspace build request"
-// @Success 200 {object} codersdk.WorkspaceBuild
-// @Router /workspaces/{workspace}/builds [post]
-// nolint:gocyclo
-func (api *API) postWorkspaceBuilds(rw http.ResponseWriter, r *http.Request) {
+func (api *API) postBuild(rw http.ResponseWriter, r *http.Request, createBuild codersdk.CreateWorkspaceBuildRequest) (codersdk.WorkspaceBuild, error) {
```

Comment:
In an effort not to duplicate logic from `postWorkspaceBuilds`, I've extracted the meat of that function and stuck it in this new helper, `postBuild`, which is then called from `postWorkspaceBuilds` and the new handler, `restartWorkspace`. I'm wondering if there's a better way to extract this logic. The way I'm handling errors (basically logging twice) seems stylistically incorrect.